### ISSUE TITLE
Error when editing user

### ISSUE THREAD

**Issue by jvalleroy** [freedombox/Plinth#449]
I created a new user named tester2, then tried to add that user to the wiki group.
```
[2016-05-01 17:56:09,071] plinth.actions ERROR    Error executing command - ['sudo', '-n', '/usr/share/plinth/actions/ssh', 'set-keys', '--username', 'tester2', '--keys', ''], , Traceback (most recent call last):
  File "/usr/share/plinth/actions/ssh", line 99, in <module>
    main()
  File "/usr/share/plinth/actions/ssh", line 95, in main
    subcommand_method(arguments)
  File "/usr/share/plinth/actions/ssh", line 73, in subcommand_set_keys
    subprocess.check_call(['mkhomedir_helper', user])
  File "/usr/lib/python3.5/subprocess.py", line 581, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['mkhomedir_helper', 'tester2']' returned non-zero exit status 8
```

**Comment by Diet-mar** [comment]
Not sure if my error is the same. Trying to edit an existing user I get "Bad username" (note, the real user-name was replaced by "removedusername"):
[2016-05-01 18:37:02,537] plinth.actions INFO     Executing command - ['sudo', '-n', '/usr/share/plinth/actions/ssh', 'get-keys', '--username', 'removedusername']
[2016-05-01 18:37:03,177] plinth.actions ERROR    Error executing command - ['sudo', '-n', '/usr/share/plinth/actions/ssh', 'get-keys', '--username', 'removedusername'], Bad username
, 
[2016-05-01 18:37:03,181] django.request ERROR    Internal Server Error: /plinth/sys/users/removedusername/edit/
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/django/core/handlers/base.py", line 149, in get_response
    response = self.process_exception_by_middleware(e, request)
  File "/usr/lib/python3/dist-packages/django/core/handlers/base.py", line 147, in get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/usr/lib/python3/dist-packages/django/views/generic/base.py", line 68, in view
    return self.dispatch(request, *args, **kwargs)
  File "/usr/lib/python3/dist-packages/django/views/generic/base.py", line 88, in dispatch
    return handler(request, *args, **kwargs)
  File "/usr/lib/python3/dist-packages/django/views/generic/edit.py", line 275, in get
    return super(BaseUpdateView, self).get(request, *args, **kwargs)
  File "/usr/lib/python3/dist-packages/django/views/generic/edit.py", line 213, in get
    return self.render_to_response(self.get_context_data())
  File "/usr/lib/python3/dist-packages/plinth/modules/users/views.py", line 43, in get_context_data
    context = super(ContextMixin, self).get_context_data(**kwargs)
  File "/usr/lib/python3/dist-packages/django/views/generic/edit.py", line 122, in get_context_data
    kwargs['form'] = self.get_form()
  File "/usr/lib/python3/dist-packages/django/views/generic/edit.py", line 74, in get_form
    return form_class(**self.get_form_kwargs())
  File "/usr/lib/python3/dist-packages/plinth/modules/users/views.py", line 83, in get_form_kwargs
    kwargs = super(UserUpdate, self).get_form_kwargs()
  File "/usr/lib/python3/dist-packages/django/views/generic/edit.py", line 167, in get_form_kwargs
    kwargs = super(ModelFormMixin, self).get_form_kwargs()
  File "/usr/lib/python3/dist-packages/django/views/generic/edit.py", line 81, in get_form_kwargs
    'initial': self.get_initial(),
  File "/usr/lib/python3/dist-packages/plinth/modules/users/views.py", line 92, in get_initial
    'ssh', ['get-keys', '--username', self.object.username]).strip()
  File "/usr/lib/python3/dist-packages/plinth/actions.py", line 118, in superuser_run
    return _run(action, options, input, async, True)
  File "/usr/lib/python3/dist-packages/plinth/actions.py", line 180, in _run
    raise ActionError(action, output, error)
plinth.errors.ActionError: ('ssh', 'Bad username\n', '')

**Comment by jvalleroy** [comment]
Partially fixed in #552. Here's another log that shows (another?) error that led to a 500 error page:
```
Traceback (most recent call last):
  File "/usr/local/lib/python3.5/dist-packages/Django-1.10-py3.5.egg/django/core/handlers/exception.py", line 39, in inner
    response = get_response(request)
  File "/usr/local/lib/python3.5/dist-packages/Django-1.10-py3.5.egg/django/core/handlers/base.py", line 249, in _legacy_get_response
    response = self._get_response(request)
  File "/usr/local/lib/python3.5/dist-packages/Django-1.10-py3.5.egg/django/core/handlers/base.py", line 187, in _get_response
    response = self.process_exception_by_middleware(e, request)
  File "/usr/local/lib/python3.5/dist-packages/Django-1.10-py3.5.egg/django/core/handlers/base.py", line 185, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/usr/local/lib/python3.5/dist-packages/Django-1.10-py3.5.egg/django/views/generic/base.py", line 68, in view
    return self.dispatch(request, *args, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/Django-1.10-py3.5.egg/django/views/generic/base.py", line 88, in dispatch
    return handler(request, *args, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/Django-1.10-py3.5.egg/django/views/generic/edit.py", line 240, in post
    return super(BaseUpdateView, self).post(request, *args, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/Django-1.10-py3.5.egg/django/views/generic/edit.py", line 183, in post
    return self.form_valid(form)
  File "/usr/local/lib/python3.5/dist-packages/Django-1.10-py3.5.egg/django/contrib/messages/views.py", line 11, in form_valid
    response = super(SuccessMessageMixin, self).form_valid(form)
  File "/usr/local/lib/python3.5/dist-packages/Django-1.10-py3.5.egg/django/views/generic/edit.py", line 162, in form_valid
    self.object = form.save()
  File "/usr/local/lib/python3.5/dist-packages/Plinth-0.9.4-py3.5.egg/plinth/modules/users/forms.py", line 209, in save
    '--keys', self.cleaned_data['ssh_keys'].strip()])
  File "/usr/local/lib/python3.5/dist-packages/Plinth-0.9.4-py3.5.egg/plinth/actions.py", line 118, in superuser_run
    return _run(action, options, input, async, True)
  File "/usr/local/lib/python3.5/dist-packages/Plinth-0.9.4-py3.5.egg/plinth/actions.py", line 180, in _run
    raise ActionError(action, output, error)
plinth.errors.ActionError: ('ssh', 'Username not found\n', '')
```

**Comment by SunilMohanAdapa** [comment]
@jvalleroy, can you confirm that #552 fixed the 'Username not found' stack strace?

**Comment by jvalleroy** [comment]
@SunilMohanAdapa It looks like there is still an issue when renaming a user.

**Comment by jvalleroy** [comment]
Fixed in #564.
